audioread/ffdec.py

```diff
@@ -73,6 +75,18 @@ def run(self):
                 # Stream closed (EOF).
                 break
 
+class WriterThread(threading.Thread):
+    """A thread that writes data to a filehandle
+    """
+    def __init__(self, fh, audio=None):
+        super(WriterThread, self).__init__()
+        self.fh = fh
+        self.audio = audio
+        self.daemon = True
+
+    def run(self):
+        self.fh.write(self.audio.read())
```

Comment:
Good call. The loop should probably exit when the file has no more data to read (i.e., when `read` returns an empty string).